Tests/HTTPHeadersTests.swift

```diff
@@ -0,0 +1,134 @@
+        headersCreatedManually.update(name: "Key", value: "")
+        headersCreatedManually.update(name: "KEY", value: "")
+
+        // When, Then
+        XCTAssertEqual(headersFromDictionaryLiteral.count, 1)
+        XCTAssertEqual(headersFromDictionary.count, 1)
+        XCTAssertEqual(headersFromArrayLiteral.count, 1)
+        XCTAssertEqual(headersFromArray.count, 1)
+        XCTAssertEqual(headersCreatedManually.count, 1)
+    }
+
+    func testHeadersPreserveOrderOfInsertion() {
+        // Given
+        let headersFromDictionaryLiteral: HTTPHeaders = ["c": "", "a": "", "b": ""]
+        // Dictionary initializer can't preserve order.
+        let headersFromArrayLiteral: HTTPHeaders = [HTTPHeader(name: "b", value: ""),
+                                                    HTTPHeader(name: "a", value: ""),
+                                                    HTTPHeader(name: "c", value: "")]
+        let headersFromArray = HTTPHeaders([HTTPHeader(name: "b", value: ""),
+                                            HTTPHeader(name: "a", value: ""),
+                                            HTTPHeader(name: "c", value: "")])
+        var headersCreatedManually = HTTPHeaders()
+        headersCreatedManually.update(HTTPHeader(name: "c", value: ""))
+        headersCreatedManually.update(name: "b", value: "")
+        headersCreatedManually.update(name: "a", value: "")
+
+        // When
+        let dictionaryLiteralNames = headersFromDictionaryLiteral.map { $0.name }
+        let arrayLiteralNames = headersFromArrayLiteral.map { $0.name }
+        let arrayNames = headersFromArray.map { $0.name }
+        let manualNames = headersCreatedManually.map { $0.name }
+
+        // Then
+        XCTAssertEqual(dictionaryLiteralNames, ["c", "a", "b"])
+        XCTAssertEqual(arrayLiteralNames, ["b", "a", "c"])
+        XCTAssertEqual(arrayNames, ["b", "a", "c"])
+        XCTAssertEqual(manualNames, ["c", "b", "a"])
+    }
+
+    func testHeadersCanBeProperlySortedByName() {
+        // Given
+        let headers: HTTPHeaders = ["c": "", "a": "", "b": ""]
+
+        // When
+        let sortedHeaders = headers.sorted()
+
+        // Then
+        XCTAssertEqual(headers.map { $0.name }, ["c", "a", "b"])
+        XCTAssertEqual(sortedHeaders.map { $0.name }, ["a", "b", "c"])
+    }
+
+    func testHeadersCanInsensitivelyGetAndSetThroughSubscript() {
+        // Given
+        var headers: HTTPHeaders = ["c": "", "a": "", "b": ""]
+
+        // When
+        headers["C"] = "c"
+        headers["a"] = "a"
+        headers["b"] = "b"
+
+        // Then
+        XCTAssertEqual(headers["c"], "c")
+        XCTAssertEqual(headers.map { $0.value }, ["c", "a", "b"])
+        XCTAssertEqual(headers.count, 3)
+    }
+
+    func testHeadersPreserveLastFormAndValueOfAName() {
+        // Given
+        var headers: HTTPHeaders = ["c": "a"]
+
+        // When
+        headers["C"] = "c"
+
```

Comment:
Extra newline.

Reply:
Fixed in [aaa3e6a](
https://github.com/Alamofire/Alamofire/pull/2629/commits/aaa3e6a10d2b14ad0dbc049dc5aa3b1b9bb8fa2e).